python/cudf/cudf/core/dataframe.py

```diff
@@ -260,11 +270,89 @@ def __init__(self, data=None, index=None, columns=None, dtype=None):
         if dtype:
             self._data = self.astype(dtype)._data
 
+    def _get_union_of_indices(self, indexes):
+        if len(indexes) == 1:
+            return indexes[0]
+        else:
+            merged_index = cudf.core.Index._concat(indexes)
+            merged_index = merged_index.drop_duplicates()
+            _, inds = merged_index._values.sort_by_values()
+            return merged_index.take(inds)
+
+    def _get_union_of_series_names(self, series_list):
+        names_list = []
+        unnamed_count = 0
+        for idx, series in enumerate(series_list):
+            if series.name is None:
+                names_list.append(f"Unnamed {unnamed_count}")
+                unnamed_count += 1
+            else:
+                names_list.append(series.name)
+        if unnamed_count == len(series_list):
+            names_list = [*range(len(series_list))]
+
+        return names_list
+
+    def _init_from_series_list(self, data, columns, index):
+        dataframe_columns = self._get_union_of_indices([d.index for d in data])
```

Comment:
The `_get_union_of_indices` function doesn't throw if there's duplicate values in the passed in objects, but Pandas expects to throw in that situation.

Need to handle it somewhere here. I believe we have similar code for trying to handle situations like these in the dictionary init for DataFrame as well as concat code where `axis=1`. cc @shwina @millerhooks who are familiar with those sections of code.

Reply:
Also need to handle the situation where there's duplicates, but the indexes are equal which is allowed.

Reply:
By duplicates, did you mean this kind of a scenario?

```python
>>> import pandas as pd
>>> s = pd.Series([1, 2, 3], index=['a', 'b', 'c'])
>>> pd.DataFrame([s, s])
   a  b  c
0  1  2  3
1  1  2  3
```

Reply:
I mean like this:
```
pd.DataFrame([pd.Series([1,2], index=['a', 'a'])])
```

Reply:
Thanks Keith